sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -220,7 +222,16 @@ def _batch_send(
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()
+            parts = response.parts()
+            if raise_on_any_failure:
+                failures = [p for p in parts if not 200 <= p.status_code < 300]
+                if failures:
+                    error = PartialBatchErrorException(
+                        message="There is a partial failure in the batch operation.",
+                        response=response, parts=iter(parts)
```

Comment:
I would assume it's consumed already and put a list in the exception, for the sake of simplicity.

@johanste if you have feelings here

Reply:
Nope. That sounds good.